### PR TITLE
Initial OpenTelemetry setup

### PR DESCRIPTION
**Pull request by myieye**

### CODE REVIEW

**Review by longrunningprocess — COMMENTED** [review]
Thanks Tim, amazing how you were able to stitch all of this together, looks quite complicated!


let's pair on this when you get a chance.  I'd like to chat through organization of this capability, naming and whether a layout load might give you what you want rather the the client and server side hooks.

File: backend/LexBoxApi/Controllers/UserController.cs [hunk 1]
```diff
@@ -55,6 +61,7 @@ public async Task<ActionResult<LexAuthUser>> RegisterAccount(RegisterAccountInpu
             PasswordHash = PasswordHashing.HashPassword(accountInput.PasswordHash, salt, true),
             IsAdmin = false
         };
+        registerActivity?.AddTag("app.user.id", userEntity.Id);
         _lexBoxDbContext.Users.Add(userEntity);
         await _lexBoxDbContext.SaveChangesAsync();
```
Comment (hahn-kev) [inline backend/LexBoxApi/Controllers/UserController.cs:66]: if there's an exception here where does that get recorded? do we have to catch that and record it in the activity? or will that happen automatically?

File: backend/LexBoxApi/Otel/OtelKernel.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        );
+    }
+
+    private static void EnrichWithUserId(this Activity activity, HttpContext httpContext)
+    {
+        var claimsPrincipal = httpContext.User;
+        var user = LexAuthUser.FromClaimsPrincipal(claimsPrincipal);
+        if (user != null)
+        {
+            activity.SetTag("app.user.id", user.Id);
```
Comment (hahn-kev) [inline backend/LexBoxApi/Otel/OtelKernel.cs:63]: wouldn't be a bad idea to record their role here too

**Comment by hahn-kev** [comment]
Looks great Tim! I'm looking forward to checking out the data in honeycomb.

File: backend/LexBoxApi/Controllers/UserController.cs (hunk 1 above)
Comment (myieye) [reply backend/LexBoxApi/Controllers/UserController.cs:66]: Exceptions are recorded automatically:
image
image

**Pull request merged**